server.js

```diff
@@ -36,11 +36,14 @@ app.get("/", (req, res, next) => {
 app.post("/query", queryAi);
 
 app.get("/task", getTasks);
-app.post("/task", createTask);
+app.post("/task", createOrUpdateTask);
+app.patch("/task", createOrUpdateTask);
```

Comment:
Following my comment on `taskHandler.js`, we should probably add the id param here like: `task/:id`.